IBAnimatable/BackgroundImageDesignable.swift

```diff
@@ -0,0 +1,61 @@
+//
+//  Created by phimage on 25/03/2017.
+//  Copyright © 2017 IBAnimatable. All rights reserved.
+//
+
+import Foundation
+
+import UIKit
+
+/// Protocol for designing background image
+public protocol BackgroundImageDesignable {
+
+  /**
+   * The background image
+   */
+  var backgroundImage: UIImage? { get set }
+
+}
+
+/// Protocol for designing background
+public protocol BackgroundDesignable: class {
+
+  /**
+   * The background view
+   */
+  var backgroundView: UIView? { get set }
+
+}
+
+extension UITableViewCell: BackgroundDesignable {}
+extension UITableViewHeaderFooterView: BackgroundDesignable {}
+extension UITableView: BackgroundDesignable {}
+extension UICollectionViewCell: BackgroundDesignable {}
+extension UICollectionView: BackgroundDesignable {}
+
+fileprivate typealias BackgroundImageView = AnimatableImageView
+
+public extension BackgroundImageDesignable where Self: BackgroundDesignable {
+  public func configureBackgroundImage() {
+    if let image = backgroundImage {
+      if let imageView = self.backgroundView as? AnimatableImageView {
```

Comment:
self is not mandatory (there is no closure)
I put `self` when possible to avoid some issue when renaming stuff et local variable become class variable etc...  
But there is poor or con, and swift lint do not have this rule yet
https://github.com/realm/SwiftLint/issues/321

Like I do in previous PR I will remove it